hermeznetwork/hermez-node
# Add CLI command to downgrade SQL schema

Closes #794 
Closes #741 

### What does this PR does?


- Add a CLI option to revert SQL migrations (migrate down)
- Add test utilities for testing SQL migrations
- Add test for the existing migration (`db/migrations/0002.sql`)

### How to test?


- Migrations are tested with unit test
- CLI option can be tested as follows:

1. Deploy a node in coordinator mode (using this branch)
2. Send at least one tx to the pool
3. Stop the node
4. Run the command `heznode migratesqldown -mode=sync -cfg=<your config file> -nMigrations=1`
5. Check that the DB has no `item_id` column and the txs you sent are still there
6. Run again the node, this time sing a version that didn't had the 2nd migration file (for instance `v1.0.0`)
7. Check that the node behaves well

## Checklist

These are the criteria that every PR should meet, please check them off as you
review them:
